cross-chain/solana/programs/tbtc/src/lib.rs

```diff
@@ -0,0 +1,357 @@
+use anchor_lang::prelude::*;
+use anchor_spl::{
+    associated_token::AssociatedToken,
+    // mint,
+    token::{mint_to, TokenAccount, Mint as SplMint, MintTo as SplMintTo, Token as SplToken},
+};
+
+declare_id!("HksEtDgsXJV1BqcuhzbLRTmXp5gHgHJktieJCtQd3pG");
+
+#[program]
+pub mod tbtc {
+    use super::*;
+    
+    pub fn initialize(ctx: Context<Initialize>) -> Result<()> {
+        let tbtc = &mut ctx.accounts.tbtc; 
+        tbtc.authority = ctx.accounts.authority.key();
+        tbtc.token_mint = ctx.accounts.tbtc_mint.key();
+        tbtc.token_bump = *ctx.bumps.get("tbtc_mint").unwrap();
+        tbtc.minters = 0;
+        tbtc.guardians = 0;
+        tbtc.paused = false;
+        Ok(())
+    }
+
+    pub fn add_minter(ctx: Context<AddMinter>) -> Result<()> {
+        let minter_info = &mut ctx.accounts.minter_info;
+        minter_info.minter = ctx.accounts.minter.key();
+        minter_info.bump = *ctx.bumps.get("minter_info").unwrap();
+
+        ctx.accounts.tbtc.minters += 1;
+        emit!(MinterAdded { minter: ctx.accounts.minter.key() });
+        Ok(())
+    }
+
+    pub fn remove_minter(ctx: Context<RemoveMinter>, minter: Pubkey) -> Result<()> {
+        // require!(ctx.accounts.tbtc.minters > 0, TbtcError::NoMinters);
+        ctx.accounts.tbtc.minters -= 1;
+        emit!(MinterRemoved { minter: minter });
+        Ok(())
+    }
+
+    pub fn add_guardian(ctx: Context<AddGuardian>) -> Result<()> {
+        let guardian_info = &mut ctx.accounts.guardian_info;
+        guardian_info.guardian = ctx.accounts.guardian.key();
+        guardian_info.bump = *ctx.bumps.get("guardian_info").unwrap();
+
+        ctx.accounts.tbtc.guardians += 1;
+        emit!(GuardianAdded { guardian: ctx.accounts.guardian.key() });
+        Ok(())
+    }
+
+    pub fn remove_guardian(ctx: Context<RemoveGuardian>, guardian: Pubkey) -> Result<()> {
+        // require!(ctx.accounts.tbtc.guardians > 0, TbtcError::NoGuardians);
+        ctx.accounts.tbtc.guardians -= 1;
+        emit!(GuardianRemoved { guardian: guardian });
+        Ok(())
+    }
+
+    pub fn setup_mint(_ctx: Context<SetupMint>) -> Result<()> {
+        msg!("setup mint complete\n\n\n");
+        Ok(())
+    }
+
+    pub fn mint(ctx: Context<Mint>, amount: u64) -> Result<()> {
+        msg!("start mint\n\n\n");
+        let seed = b"tbtc-mint";
+        let key_seed = ctx.accounts.tbtc.key();
+        let mint_bump = ctx.accounts.tbtc.token_bump;
+        let signer: &[&[&[u8]]] = &[&[seed, key_seed.as_ref(), &[mint_bump]]];
+
+        let cpi_ctx = CpiContext::new_with_signer(
+            ctx.accounts.token_program.to_account_info(),
+            SplMintTo {
+                mint: ctx.accounts.tbtc_mint.to_account_info(),
+                to: ctx.accounts.recipient_account.to_account_info(),
+                authority: ctx.accounts.tbtc_mint.to_account_info(),
+            },
+            signer,
+        );
+        mint_to(cpi_ctx, amount)?;
```

Comment:
No need to use `?` here and then end the instruction with `Ok`. You can end the instruction by doing:
```
mint_to(cpi_ctx, amount)
```